arrow/src/main/java/org/apache/iceberg/arrow/ArrowSchemaUtil.java

```diff
@@ -51,94 +54,131 @@ private ArrowSchemaUtil() {}
   public static Schema convert(final org.apache.iceberg.Schema schema) {
     ImmutableList.Builder<Field> fields = ImmutableList.builder();
 
-    for (NestedField f : schema.columns()) {
-      fields.add(convert(f));
+    for (NestedField field : schema.columns()) {
+      fields.add(TypeUtil.visit(field.type(), new IcebergToArrowTypeConverter(field)));
     }
-
```

Comment:
Nit: why remove this line?
The style guide requires a newline after every block

Reply:
Done, but I thought we used the Google Java style, which [doesn't](https://google.github.io/styleguide/javaguide.html#s4.6-whitespace) **require** a new line after every block.